cgo/compare.c

```diff
@@ -0,0 +1,370 @@
+/*
+ * Copyright 2021 Matrix Origin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "mo_impl.h"
+
+
+#define	Type_BOOL     10
+#define	Type_INT8     20
+#define	Type_INT16    21
+#define	Type_INT32    22
+#define	Type_INT64    23
+#define	Type_INT128   24
+#define	Type_UINT8    25
+#define	Type_UINT16   26
+#define	Type_UINT32   27
+#define	Type_UINT64   28
+#define	Type_UINT128  29
+#define	Type_FLOAT32  30
+#define	Type_FLOAT64  31
+
+// Time
+#define Type_DATE       50
+#define Type_DATETIME   52
+#define Type_TIMESTAMP  53
+
+
+/*
+ * Equal operator (=)
+ */
+#define COMPARE_EQ(TGT, A, B)                                 \
+    TGT = ((A) == (B))
+
+
+/*
+ * Not Equal operator (<>)
+ */
+#define COMPARE_NE(TGT, A, B)                                 \
+    TGT = ((A) != (B))
+
+
+/*
+ * great than operator (>)
+ */
+#define COMPARE_GT(TGT, A, B)                                 \
+    TGT = ((A) > (B))
+
+
+/*
+ * great equal operator (>=)
+ */
+#define COMPARE_GE(TGT, A, B)                                 \
+    TGT = ((A) >= (B))
+
+
+/*
+ * less than operator (<)
+ */
+#define COMPARE_LT(TGT, A, B)                                 \
+    TGT = ((A) < (B))
+
+
+/*
+ * less equal operator (<=)
+ */
+#define COMPARE_LE(TGT, A, B)                                 \
+    TGT = ((A) <= (B))
+
+
+/*
+ * bool compare operator
+ */
```

Comment:
Are you sure that the value of the bool in Golang is same as the one in CLang?
So, the  '==' and  '!=' for the equal or not equal operators of the bool needs to be considered.

Reply:
OK， I use the C language style to compare bool types